File: base/src/main/java/com/lotteryviewer/base/BaseApplication.java
package com.lotteryviewer.base;

import android.app.Application;

/**
 * @Author: duke
 * @DateTime: 2021-03-17 10:53:57
 * @Description: 功能描述：
 */
public class BaseApplication {

    private static Application INSTANCE;

    public static void injecctApp(Application application) {
        if (application == null) {
            throw new NullPointerException("Application 为空");
        }
        INSTANCE = application;
        ActivityStack.init(application);
    }

    public static Application get() {
        if (INSTANCE == null) {
            throw new NullPointerException("Application 为空");
        }
        return INSTANCE;
    }

}

File: base/src/main/java/com/lotteryviewer/base/app/BaseApplication.kt
package com.lotteryviewer.base.app

import android.app.Application
import com.lotteryviewer.base.app.ActivityStack.init

/**
 * @Author: duke
 * @DateTime: 2021-03-17 10:53:57
 * @Description: 功能描述：
 */
object BaseApplication {

    private var mApplication: Application? = null

    private fun setApplication(application: Application) {
        mApplication = application
    }

    fun get(): Application? {
        if (mApplication != null) {
            return mApplication
        }
        throw NullPointerException("Application 为空")
    }

    fun init(application: Application) {
        setApplication(application)
        init(get())
        CrashExceptionHandler.instance?.setDefaultUnCrashExceptionHandler()
    }

}